driver-core/src/main/com/mongodb/internal/connection/SplittablePayload.java

```diff
@@ -74,11 +92,11 @@
      * Create a new instance
      *
      * @param payloadType the payload type
-     * @param payload the payload
+     * @param writeRequestWithIndexes the writeRequests
      */
-    public SplittablePayload(final Type payloadType, final List<BsonDocument> payload) {
+    public SplittablePayload(final Type payloadType, final List<WriteRequestWithIndex> writeRequestWithIndexes) {
```

Comment:
The major change here was to include the `WriteRequestWithIndex` into the `SplittablePayload` so we use the correct `BsonWriter` when doing inserts.